src/MotionHandler.moon

```diff
@@ -171,21 +171,56 @@ class MotionHandler
-						message = "There is a malformed \\fade you must fix.\n\\fade requires 7 integer arguments.\nLine: #{.number}, tag: \\fade(#{fade})."
-						if fade\match("(%d+),(%d+)")
-							message ..= "\nPerhaps you meant to use \\fad."
-						log.windowError message
-
-					t1, t2, t3, t4 = tonumber( t1 ), tonumber( t2 ), tonumber( t3 ), tonumber( t4 )
-					-- beautiful.
-					t1 -= timeDelta
-					t2 -= timeDelta
-					t3 -= timeDelta
-					t4 -= timeDelta
-					("\\fade(%s,%s,%s,%d,%d,%d,%d)")\format a1, a2, a3, t1, t2, t3, t4
+				fadeTag = tags.allTags.fade
+				if @options.main.killTrans
+					-- multiplies every alpha tag by a scaling factor determined by the fade envelope
+					local fade
+					newText = newText\gsub "({.-})", ( tagBlock ) -> tagBlock\gsub fadeTag.pattern, ( value ) ->
+						fade = fadeTag\convert value
+						-- TODO: broken when not all tags exist, can this functionality be moved to Aegisub-Motion:getMissingTags?
+						if not tagBlock\find "\\[1234]?a[lpha]-&H%x%x&"
+							return "\\alpha&H00&"
+						else
+							return ""
+
+					if fade
+						local fadeFactor
+						-- Piecewise function done with logicals :)
+						-- probably should go in fadeTag\interpolate, but I dunno how that whole section is supposed to work
+						f = { k, tonumber v for k, v in pairs fade }
+						fadeFactor = (
+							-- x < t1: 			a1
+							(timeDelta < f["t1"]) and 	f["a1"] or 
+							-- t1 <= x < t2:	a1 + (a2 - a1) * (x - t1) / (t2 - t1)
+							(timeDelta < f["t2"]) and 	f["a1"] + (f["a2"] - f["a1"]) * (timeDelta - f["t1"]) / (f["t2"] - f["t1"]) or
+							-- t2 <= x < t3		a2
+							(timeDelta < f["t3"]) and 	f["a2"] or
+							-- t3 <= x < t4		a2 + (a3 - a2) * (x - t3) / (t4 - t3)
+							(timeDelta < f["t4"]) and 	f["a2"] + (f["a3"] - f["a2"]) * (timeDelta - f["t3"]) / (f["t4"] - f["t3"]) or
+							-- x >= t4:			a3
+							 							f["a3"]
+						)
+						-- factor between 0 and 1 representing opacity of fade line
+						fadeFactor = (255 - fadeFactor) / 255
+
+						-- apply the opacity factor to all alpha tags
+						newText = newText\gsub "({.-})", ( tagBlock ) -> tagBlock\gsub "(\\[1234]?a[lpha]-)&H(%x%x)&", ( alpha, value ) ->
+								value = Math.round( 255 - ( fadeFactor * (255 - tonumber value, 16) ) )
+								return alpha .. "&H%x&"\format value
+
+				else
+					-- modified each fade tag in every override block
+					newText = newText\gsub "({.-})", ( tagBlock ) -> tagBlock\gsub fadeTag.pattern, ( value ) ->
+						fade = fadeTag\convert value
+						-- Erroneous fade tags are being ignored and left sitting around as long as they doesn't have 2 or 7 arguments.
+						-- if t1 == nil
+						-- 	message = "There is a malformed \\fade you must fix.\n\\fade requires 7 integer arguments.\nLine: #{.number}, tag: \\fade(#{fade})."
+						-- 	if fade\match("(%d+),(%d+)")
+						-- 		message ..= "\nPerhaps you meant to use \\fad."
+						-- 	log.windowError message
+						for i, field in ipairs { 4, 5, 6, 7 } -- t1, t2, t3, t4
```

Comment:
you can avoid the overhead of allocating a table and the ipairs iterator call here by just using `for field = 4,7`